ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullQueryRewriter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.planner.plan;
+
+import com.google.common.collect.Lists;
+import io.confluent.ksql.engine.rewrite.ExpressionTreeRewriter;
+import io.confluent.ksql.engine.rewrite.ExpressionTreeRewriter.Context;
+import io.confluent.ksql.engine.rewrite.StatementRewriteForMagicPseudoTimestamp;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression.Type;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.InPredicate;
+import io.confluent.ksql.execution.expression.tree.LogicalBinaryExpression;
+import io.confluent.ksql.execution.expression.tree.VisitParentExpressionVisitor;
+import java.util.Optional;
+
+public final class PullQueryRewriter {
```

Comment:
It's just an arbitrary split to avoid having a mega class with many rewriters and the hope that others in the future might want to use `LogicalRewriter` outside of pull queries.  The stuff here is pull query specific and I didn't want to mix that with generic logic.  Do you disagree with that split?

Reply:
Ah okay, that sounds good to me. I was just not clear what's the splitting discipline on which func goes to which class.